File: medium/3737_count_subarrays_with_majority_element_i/solution.cpp
#include <vector>

class Solution {
public:
  int countMajoritySubarrays(std::vector<int> &nums, int target) {
    const size_t n = nums.size();

    int count = 0;

    for (size_t i = 0; i < n; ++i) {
      int targetCount = 0;

      for (size_t j = i; j < n; ++j) {
        if (nums[j] == target)
          ++targetCount;
        else
          --targetCount;

        if (targetCount > 0)
          ++count;
      }
    }

    return count;
  }
};

File: medium/3737_count_subarrays_with_majority_element_i/solution.py
from typing import List


class Solution:
    def countMajoritySubarrays(self, nums: List[int], target: int) -> int:
        n = len(nums)

        prefixCounts = [0] * (len(nums) + 1)

        for i in range(1, n + 1):
            prefixCounts[i] = prefixCounts[i - 1] + (nums[i - 1] == target)

        count = 0

        for i in range(n):
            for j in range(i, n):
                if 2 * (prefixCounts[j + 1] - prefixCounts[i]) > j - i + 1:
                    count += 1

        return count
